addons/account/models/account_move.py

```diff
@@ -3729,6 +3729,18 @@ def name_get(self):
             result.append((line.id, name))
         return result
 
+    @api.model
+    def invalidate_cache(self, fnames=None, ids=None):
+        # Invalidate cache of related moves
+        if not fnames or 'move_id' in fnames:
+            f = self._fields['move_id']
+            line2move = self.env.cache._data.get(f)
+            if line2move:
+                move_ids = [move_id for line_id, move_id in line2move.items()
+                            if not ids or line_id in ids]
+                self.env['account.move'].browse(move_ids).invalidate_cache()
```

Comment:
You will invalidate all the cache by doing this (`invalidate_cache` doesn't depend of self). 
Maybe : `self.env['account.move'].invalidate_cache(ids=move_ids)`